### PR TITLE
Update `DefaultResourcesNameStep`

### PR DESCRIPTION
**Pull request by MicroFish91**
Closes https://github.com/microsoft/vscode-azurecontainerapps/issues/611
Closes https://github.com/microsoft/vscode-azurecontainerapps/issues/603

- Removes auto-naming of resources based on the workspace name and always prompts for a name
- Refactor to leverage each step's `validateInput` (now static methods)
- Remove tests that are no longer necessary since logic has become much simpler 🧹

### CODE REVIEW

**Comment by MicroFish91** [comment]
@nturinski Double check this one please

File: src/commands/deployWorkspaceProject/internal/DefaultResourcesNameStep.ts [hunk 1]
```diff
@@ -4,83 +4,72 @@
...
-            ext.outputChannel.appendLog(localize('usingWorkspaceName', 'Using workspace name "{0}" as the default for remaining resource creation.', workspaceName));
-        }
+        ext.outputChannel.appendLog(localize('usingResourceName', 'User provided the new resource name "{0}" as the default for resource creation.', resourceName))
 
-        !context.resourceGroup && (context.newResourceGroupName = workspaceName);
-        !context.managedEnvironment && (context.newManagedEnvironmentName = workspaceName);
-        !context.registry && (context.newRegistryName = await RegistryNameStep.tryGenerateRelatedName(context, workspaceName));
-        !context.containerApp && (context.newContainerAppName = workspaceName);
-        context.imageName = ImageNameStep.getTimestampedImageName(context.containerApp?.name || workspaceName);
+        !context.resourceGroup && (context.newResourceGroupName = resourceName);
```
Comment (nturinski) [inline src/commands/deployWorkspaceProject/internal/DefaultResourcesNameStep.ts:27]: I still suggest we should add a suffix for each resource type.

`resourceName-rg`
`resourceName-me` or `resourceName-env`
`resourceName-app` or `resourceName-aca`

I think when you see them all in the resource group, it makes it a lot easier to distinguish what each resource is. I notice other tooling will do that as well.

**Review by nturinski — APPROVED** [review]
Rest looks good to me. So fresh, so clean

File: src/commands/deployWorkspaceProject/internal/DefaultResourcesNameStep.ts (hunk 1 above)
Comment (MicroFish91) [reply src/commands/deployWorkspaceProject/internal/DefaultResourcesNameStep.ts:27]: I like that idea, let me do that in a separate PR after we discuss what that convention should look like

**Pull request merged**